src/meal/admin.py

```diff
@@ -30,32 +33,67 @@ class ComponentAdmin(admin.ModelAdmin):
         ComponentIngredientInline
     ]
     list_display = ('name', 'component_group')
-    search_fields = ['name', 'component_group']
+    search_fields = ('name',)
 
     def get_ordering(self, request):
         return [Lower('name')]
 
+    def get_search_results(self, request, queryset, search_term):
+        # search for group choices in their display value instead of DB value
+        queryset, use_distinct = super().get_search_results(
+            request, queryset, search_term)
+        search_list = [choice[0] for
+                       choice in COMPONENT_GROUP_CHOICES
+                       if search_term in choice[1]]
```

Comment:
Could we perform the `in` in a case-insensitive manner for the group choices? This is to conform with Django's default behaviour:
![image](https://cloud.githubusercontent.com/assets/8630726/23759571/24fb31b0-04bb-11e7-885d-f3d020f5c18e.png)

And on the group choice column, the case matters:
![image](https://cloud.githubusercontent.com/assets/8630726/23759591/3196ba16-04bb-11e7-8c22-dd8a8aa68065.png)

![image](https://cloud.githubusercontent.com/assets/8630726/23759631/4e4a3d36-04bb-11e7-95c9-93bdfda60133.png)